### PR TITLE
rocmPackages_5: disallow further version upgrades, fixup links

### PR DESCRIPTION
**Pull request by Madouura**
## Description of changes
Tracking: #197885
Preparation PR for ROCm 6.0.0.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Madouura** [comment]
Result of `nixpkgs-review pr 274980` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages marked as broken and skipped:</summary>
  <ul>
    <li>rocmPackages.llvm.flang</li>
    <li>rocmPackages.llvm.flang.doc</li>
    <li>rocmPackages.llvm.flang.info</li>
    <li>rocmPackages.llvm.flang.man</li>
    <li>rocmPackages.rdc</li>
    <li>rocmPackages.rdc.doc</li>
    <li>rocmPackages_5.llvm.flang</li>
    <li>rocmPackages_5.llvm.flang.doc</li>
    <li>rocmPackages_5.llvm.flang.info</li>
    <li>rocmPackages_5.llvm.flang.man</li>
    <li>rocmPackages_5.rdc</li>
    <li>rocmPackages_5.rdc.doc</li>
  </ul>
</details>
<details>
  <summary>60 packages built:</summary>
  <ul>
    <li>blender-hip</li>
    <li>magma (magma-hip ,magma_2_7_2)</li>
    <li>magma_2_6_2</li>
    <li>opensyclWithRocm</li>
    <li>python310Packages.torchWithRocm</li>
    <li>python310Packages.torchWithRocm.cxxdev</li>
    <li>python310Packages.torchWithRocm.dev</li>
    <li>python310Packages.torchWithRocm.dist</li>
    <li>python310Packages.torchWithRocm.lib</li>
    <li>python311Packages.torchWithRocm</li>
    <li>python311Packages.torchWithRocm.cxxdev</li>
    <li>python311Packages.torchWithRocm.dev</li>
    <li>python311Packages.torchWithRocm.dist</li>
    <li>python311Packages.torchWithRocm.lib</li>
    <li>rocmPackages.clr (rocmPackages_5.clr)</li>
    <li>rocmPackages.clr.icd (rocmPackages_5.clr.icd)</li>
    <li>rocmPackages.composable_kernel (rocmPackages_5.composable_kernel)</li>
    <li>rocmPackages.hipblas (rocmPackages_5.hipblas)</li>
    <li>rocmPackages.hipcub (rocmPackages_5.hipcub)</li>
    <li>rocmPackages.hipfft (rocmPackages_5.hipfft)</li>
    <li>rocmPackages.hiprand (rocmPackages.rocrand ,rocmPackages_5.hiprand ,rocmPackages_5.rocrand)</li>
    <li>rocmPackages.hipsolver (rocmPackages_5.hipsolver)</li>
    <li>rocmPackages.hipsparse (rocmPackages_5.hipsparse)</li>
    <li>rocmPackages.llvm.mlir (rocmPackages_5.llvm.mlir)</li>
    <li>rocmPackages.llvm.openmp (rocmPackages_5.llvm.openmp)</li>
    <li>rocmPackages.llvm.openmp.doc (rocmPackages_5.llvm.openmp.doc)</li>
    <li>rocmPackages.llvm.openmp.info (rocmPackages_5.llvm.openmp.info)</li>
    <li>rocmPackages.llvm.openmp.man (rocmPackages_5.llvm.openmp.man)</li>
    <li>rocmPackages.migraphx (rocmPackages_5.migraphx)</li>
    <li>rocmPackages.miopen (rocmPackages.miopen-hip ,rocmPackages_5.miopen ,rocmPackages_5.miopen-hip)</li>
    <li>rocmPackages.miopen-opencl (rocmPackages_5.miopen-opencl)</li>
    <li>rocmPackages.miopengemm (rocmPackages_5.miopengemm)</li>
    <li>rocmPackages.miopengemm.doc (rocmPackages_5.miopengemm.doc)</li>
    <li>rocmPackages.mivisionx (rocmPackages.mivisionx-hip ,rocmPackages_5.mivisionx ,rocmPackages_5.mivisionx-hip)</li>
    <li>rocmPackages.mivisionx-cpu (rocmPackages_5.mivisionx-cpu)</li>
    <li>rocmPackages.mivisionx-opencl (rocmPackages_5.mivisionx-opencl)</li>
    <li>rocmPackages.rccl (rocmPackages_5.rccl)</li>
    <li>rocmPackages.rocalution (rocmPackages_5.rocalution)</li>
    <li>rocmPackages.rocblas (rocmPackages_5.rocblas)</li>
    <li>rocmPackages.rocdbgapi (rocmPackages_5.rocdbgapi)</li>
    <li>rocmPackages.rocdbgapi.doc (rocmPackages_5.rocdbgapi.doc)</li>
    <li>rocmPackages.rocfft (rocmPackages_5.rocfft)</li>
    <li>rocmPackages.rocm-runtime (rocmPackages_5.rocm-runtime)</li>
    <li>rocmPackages.rocminfo (rocmPackages_5.rocminfo)</li>
    <li>rocmPackages.rocmlir (rocmPackages_5.rocmlir)</li>
    <li>rocmPackages.rocmlir-rock (rocmPackages_5.rocmlir-rock)</li>
    <li>rocmPackages.rocmlir.external (rocmPackages_5.rocmlir.external)</li>
    <li>rocmPackages.rocprim (rocmPackages_5.rocprim)</li>
    <li>rocmPackages.rocprofiler (rocmPackages_5.rocprofiler)</li>
    <li>rocmPackages.rocr-debug-agent (rocmPackages_5.rocr-debug-agent)</li>
    <li>rocmPackages.rocsolver (rocmPackages_5.rocsolver)</li>
    <li>rocmPackages.rocsparse (rocmPackages_5.rocsparse)</li>
    <li>rocmPackages.rocthrust (rocmPackages_5.rocthrust)</li>
    <li>rocmPackages.roctracer (rocmPackages_5.roctracer)</li>
    <li>rocmPackages.rocwmma (rocmPackages_5.rocwmma)</li>
    <li>rocmPackages.rpp (rocmPackages.rpp-hip ,rocmPackages_5.rpp ,rocmPackages_5.rpp-hip)</li>
    <li>rocmPackages.rpp-cpu (rocmPackages_5.rpp-cpu)</li>
    <li>rocmPackages.rpp-opencl (rocmPackages_5.rpp-opencl)</li>
    <li>rocmPackages.tensile (rocmPackages_5.tensile)</li>
    <li>rocmPackages.tensile.dist (rocmPackages_5.tensile.dist)</li>
  </ul>
</details>

File: pkgs/development/rocm-modules/5/clr/default.nix [hunk 1]
```diff
@@ -82,7 +82,7 @@ in stdenv.mkDerivation (finalAttrs: {
     "-DROCM_PATH=${rocminfo}"
 
     # Temporarily set variables to work around upstream CMakeLists issue
-    # Can be removed once https://github.com/ROCm-Developer-Tools/hipamd/issues/55 is fixed
+    # Can be removed once https://github.com/ROCm/hipamd/issues/55 is fixed
```
Comment (Flakebi) [inline pkgs/development/rocm-modules/5/clr/default.nix:85]: This seems to be still the old link

**Review by Flakebi — COMMENTED** [review]
Thanks!

**Comment by pbsds** [comment]
Diff LGTM, but should this `broken` criteria be added to each package, or would it make sense to somehow apply it to the whole package set?

**Comment by Madouura** [comment]
> Diff LGTM, but should this `broken` criteria be added to each package, or would it make sense to somehow apply it to the whole package set?

Applying mass breakage is something I'm planning for ``rocmPackages_6``, which I will, when time allows, backport to ``rocmPackages_5``.

**Comment by Madouura** [comment]
Only change, the ``rocm-cmake`` issue is more accurate anyway, and links to the old issue.
```diff
diff --git a/pkgs/development/rocm-modules/5/clr/default.nix b/pkgs/development/rocm-modules/5/clr/default.nix
index b5676a8256af..1a72a8d7c3fb 100644
--- a/pkgs/development/rocm-modules/5/clr/default.nix
+++ b/pkgs/development/rocm-modules/5/clr/default.nix
@@ -82,7 +82,7 @@ in stdenv.mkDerivation (finalAttrs: {
     "-DROCM_PATH=${rocminfo}"
 
     # Temporarily set variables to work around upstream CMakeLists issue
-    # Can be removed once https://github.com/ROCm/hipamd/issues/55 is fixed
+    # Can be removed once https://github.com/ROCm/rocm-cmake/issues/121 is fixed
     "-DCMAKE_INSTALL_BINDIR=bin"
     "-DCMAKE_INSTALL_INCLUDEDIR=include"
     "-DCMAKE_INSTALL_LIBDIR=lib"
```

File: pkgs/development/rocm-modules/5/clr/default.nix (hunk 1 above)
Comment (Madouura) [reply pkgs/development/rocm-modules/5/clr/default.nix]: Changed to https://github.com/ROCm/rocm-cmake/issues/121.

**Review by Flakebi — APPROVED** [review]

**Pull request merged**